### ISSUE TITLE
where should I put .txt files about annotation information

### ISSUE THREAD

**Issue by xiaohai12** [AlexeyAB/darknet#2872]
When I start to train yolov3 on my own dataset, I putted the .txt files with images , and it worked well. But when I try to test then and output the result.txt file for prediction, and it required me to put txt file in labels dictionary?

**Comment by AlexeyAB** [comment]
> But when I try to test then and output the result.txt file for prediction, 

What command do you use?

> and it required me to put txt file in labels dictionary.

Whu do you think so?

**Comment by xiaohai12** [comment]
for training: ./darknet detector train data/obj.data data/yolov3-obj.cfg backup/weights
for test: ./darknet detector test data/obj.data data/yolov3-obj.cfg backup/yolov3-obj_XXX.weights -dont_show -ext_output < data/valid.txt > result.txt

**Comment by xiaohai12** [comment]
> > But when I try to test then and output the result.txt file for prediction,
> 
> What command do you use?
> 
> > and it required me to put txt file in labels dictionary.
> 
> Whu do you think so?

for training: ./darknet detector train data/obj.data data/yolov3-obj.cfg backup/weights
for test: ./darknet detector test data/obj.data data/yolov3-obj.cfg backup/yolov3-obj_XXX.weights -dont_show -ext_output < data/valid.txt > result.txt

**Comment by AlexeyAB** [comment]
Can you attach screenshot?

**Comment by xiaohai12** [comment]
> Can you attach screenshot?
here is an example that I only put txt file in the labels dictionary.
The first image shows when I tried to train my data
![2019-04-10 09_36_11-Photos](https://user-images.githubusercontent.com/27256068/55860665-11d53f00-5b75-11e9-833b-f09e4dce6e79.png)

the second image shows when I test and output result as result.txt:
![2019-04-10 09_40_09-Photos](https://user-images.githubusercontent.com/27256068/55860757-3f21ed00-5b75-11e9-8204-8c71ae8317cf.png)
but even it mentioned that can not load images from labels dictionary, it still worked for test.

**Comment by AlexeyAB** [comment]
> here is an example that I only putted txt files in the labels dictionary.
> The first image shows when I tried to train my data.
> ![2019-04-10 09_36_11-Photos](https://user-images.githubusercontent.com/27256068/55860665-11d53f00-5b75-11e9-833b-f09e4dce6e79.png)

**You must put txt-labels in the same directory where are images.**

----

> the second image shows when I tried to test and output result as result.txt,but even it mentioned that can not load images from labels dictionary, it still worked for test.
> ![2019-04-10 09_40_09-Photos](https://user-images.githubusercontent.com/27256068/55860757-3f21ed00-5b75-11e9-8204-8c71ae8317cf.png)
   
**It isn't about your images. And isn't about training dataset.
These are images with letters that are displayed on the screen if you compiled Darknet without OpenCV. Open this images:  https://github.com/AlexeyAB/darknet/tree/master/data/labels**

**Comment by xiaohai12** [comment]
> > here is an example that I only putted txt files in the labels dictionary.
> > The first image shows when I tried to train my data.
> > ![2019-04-10 09_36_11-Photos](https://user-images.githubusercontent.com/27256068/55860665-11d53f00-5b75-11e9-833b-f09e4dce6e79.png)
> 
> **You must put txt-labels in the same directory where are images.**
> 
> > the second image shows when I tried to test and output result as result.txt,but even it mentioned that can not load images from labels dictionary, it still worked for test.
> > ![2019-04-10 09_40_09-Photos](https://user-images.githubusercontent.com/27256068/55860757-3f21ed00-5b75-11e9-8204-8c71ae8317cf.png)
> 
> **It isn't about your images. And isn't about training dataset. These are images with letters that are displayed on the screen if you compiled Darknet without OpenCV. Open this images: https://github.com/AlexeyAB/darknet/tree/master/data/labels**

yes. But when I tested the model, it seems that I need to put txt files in the labels dictionary, because the test can work when I putted txt files in the labels dictionary in the previous example

**Comment by AlexeyAB** [comment]
> yes. But **when I tested the model, it seems that I need to put txt files in the labels dictionary**, because the test can work when I putted txt files in the labels dictionary in the previous example

No.
You need not to put txt files in the labels dictionar.
You need not to have txt label files at all for testing.
You just need not to delete this folder: https://github.com/AlexeyAB/darknet/tree/master/data/labels

**Comment by xiaohai12** [comment]
> > yes. But **when I tested the model, it seems that I need to put txt files in the labels dictionary**, because the test can work when I putted txt files in the labels dictionary in the previous example
> 
> No.
> You need not to put txt files in the labels dictionar.
> You need not to have txt label files at all for `./darknet detector test` ...
> You just need not to delete this folder: https://github.com/AlexeyAB/darknet/tree/master/data/labels

Thanks! It can work well when I delete labels dictionary!
